azurerm/resource_arm_action_group.go

```diff
@@ -0,0 +1,313 @@
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.NoZeroValues,
+						},
+						"service_uri": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.NoZeroValues,
+						},
+					},
+				},
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmActionGroupCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).actionGroupsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	resGroup := d.Get("resource_group_name").(string)
+
+	shortName := d.Get("short_name").(string)
+	enabled := d.Get("enabled").(bool)
+
+	tags := d.Get("tags").(map[string]interface{})
+	expandedTags := expandTags(tags)
+
+	parameters := insights.ActionGroupResource{
+		Location: &location,
+		ActionGroup: &insights.ActionGroup{
+			GroupShortName: &shortName,
+			Enabled:        &enabled,
+		},
+		Tags: expandedTags,
+	}
+
+	if v, ok := d.GetOk("email_receiver"); ok {
+		parameters.ActionGroup.EmailReceivers = expandActionGroupEmailReceiver(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("sms_receiver"); ok {
+		parameters.ActionGroup.SmsReceivers = expandActionGroupSmsReceiver(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("webhook_receiver"); ok {
+		parameters.ActionGroup.WebhookReceivers = expandActionGroupWebHookReceiver(v.([]interface{}))
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resGroup, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating or updating action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error getting action group %s (resource group %s) after creation: %+v", name, resGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Action group %s (resource group %s) ID is empty", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmActionGroupRead(d, meta)
+}
+
+func resourceArmActionGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).actionGroupsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing action group resource ID \"%s\" during get: %+v", d.Id(), err)
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["actionGroups"]
+
+	resp, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		if response.WasNotFound(resp.Response.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error getting action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	d.Set("short_name", *resp.GroupShortName)
+	d.Set("enabled", *resp.Enabled)
```

Comment:
we can remove the pointer deference here and we should be accessing these on the `.Properties` block to match the Azure API Response